psi4/src/psi4/scfgrad/response.cc

```diff
@@ -4694,21 +4707,44 @@ void USCFDeriv::assemble_Q(std::shared_ptr<JK> jk,
         }
 
         for (int a = 0; a < nA; a++) {
-            //C_DGEMM('N','N',nso,n1occ,nso, 4.0,J[a]->pointer()[0],nso,C1op[0],n1occ,0.0,Tp[0],n1occ);
-            C_DGEMM('N','N',nso,n1occ,nso, 2.0,J[2*a]->pointer()[0],nso,C1op[0],n1occ,0.0,Tp[0],n1occ);
-            C_DGEMM('N','N',nso,n1occ,nso, 2.0,J[2*a+1]->pointer()[0],nso,C1op[0],n1occ,1.0,Tp[0],n1occ);
-            if(Kscale) {
-                C_DGEMM('N','N',nso,n1occ,nso,-Kscale,K[2*a]->pointer()[0],nso,C1op[0],n1occ,1.0,Tp[0],n1occ);
-                C_DGEMM('T','N',nso,n1occ,nso,-Kscale,K[2*a]->pointer()[0],nso,C1op[0],n1occ,1.0,Tp[0],n1occ);
+            // Now construct the matrices. Both the alpha and beta term sum the two J terms.
+            // When we're done, J[2*a] will be the running alpha result,and J[2*a+1] the beta.
+            J[2*a]->add(J[2*a+1]);
+            J[2*a]->scale(2);
+            J[2*a+1]->copy(J[2*a]);
+
+            if (Kscale) {
+                K[2*a]->add(K[2*a]->transpose());
+                K[2*a+1]->add(K[2*a+1]->transpose());
+                K[2*a]->scale(-Kscale);
+                K[2*a+1]->scale(-Kscale);
+                J[2*a]->add(K[2*a]);
+                J[2*a+1]->add(K[2*a+1]);
             }
-            if(functional_->needs_xc()) {
-                // Symmetrize the result, just to be safe
-                C_DGEMM('N','N',nso,n1occ,nso, 2.0,Vx[2*a]->pointer()[0],nso,C1op[0],n1occ,1.0,Tp[0],n1occ);
-                C_DGEMM('T','N',nso,n1occ,nso, 2.0,Vx[2*a]->pointer()[0],nso,C1op[0],n1occ,1.0,Tp[0],n1occ);
+
+            // Add DFT, if any.
+            if (functional_->needs_xc()) {
+                // Symmetrize the result, just to be safe, ans we also need to double it.
```

Comment:
```suggestion
                // Symmetrize the result, just to be safe, and we also need to double it.
```
In case you need an excuse to retriever azure